### PR TITLE
Ubuntu snap: Create sdk-tests snap to test integration with the agent snap

### PR DESCRIPTION
**Pull request by andretoyama-msft**
- Add skd snap with sdk-test binarie
 - Add script for handling multi snapcrafts files in the same repo

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -83,7 +83,9 @@ if (DO_PLATFORM_LINUX)
     # (Enable --trace-expand in cmake and look at lines from boost_headers-config.cmake and its callers.)
     # Workaround: force cmake to look in "/usr" since the /lib and /bin paths are either
     # not relevant for our usage or they are symlinks to /usr/lib and /usr/bin.
-    set(CMAKE_PREFIX_PATH "/usr")
+    if (NOT CMAKE_PREFIX_PATH)
```
Comment (shishirb-MSFT) [inline CMakeLists.txt:86]: This change is already in develop. You should update your local branch:
git fetch -p
git rebase -i origin/develop

File: build/build-snaps.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#! /bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+set -e
+
+if [ $1 == "agent" -o $1 == "sdk" ]
+then
+    echo "Building $SNAP snap..."
```
Comment (shishirb-MSFT) [inline build/build-snaps.sh:10]: Change $SNAP to $1

File: build/build-snaps.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#! /bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+set -e
+
+if [ $1 == "agent" -o $1 == "sdk" ]
```
Comment (shishirb-MSFT) [inline build/build-snaps.sh:8]: Let's use "sdk-tests" as the argument for more clarity.

File: build/build-snaps.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+then
+    echo "Building $SNAP snap..."
+
+    if [ ! -d "/snap" ]
```
Comment (shishirb-MSFT) [inline build/build-snaps.sh:12]: "/snap" is the root snap folder. Change to "./snap" (also in the mkdir call for consistency).

File: sdk-cpp/tests/test_data.cpp [hunk 5]
```diff
@@ -10,11 +10,19 @@ const uint64_t g_smallFileSizeBytes = 1837u;
 const uint64_t g_largeFileSizeBytes = 536870440u;
 const uint64_t g_prodFileSizeBytes = 25006511u;
 
+std::string downloadPath = "/var/lib/deliveryoptimization-snap-downloads-root";
+
+// #ifdef DO_BUILD_FOR_SNAP
```
Comment (shishirb-MSFT) [inline sdk-cpp/tests/test_data.cpp:15]: For the ifdef to work, you have to pass DO_BUILD_FOR_SNAP into the sdk tests compilation via sdk-cpp/tests/CMakeLists.txt. See how the DO agent cmakelists does it.

File: snapcraft-options/snapcraft-agent.yaml [hunk 6]
```diff
@@ -37,6 +37,7 @@ parts:
     override-build: |
       python3 ./build/build.py --project agent --build-for-snap
       mkdir -p ../install/bin
+      mkdir -p ../install/var/run/deliveryoptimization-agent
```
Comment (shishirb-MSFT) [inline snapcraft-options/snapcraft-agent.yaml:40]: Creating the ../install/var/run/deliveryoptimization-agent is not required I think.

File: snapcraft-options/snapcraft-agent.yaml [hunk 7]
```diff
@@ -80,7 +81,19 @@ slots:
     content: port-number
     read: [ $SNAP_DATA/var/run/deliveryoptimization-agent ]
 
+  # Share a folder containing sdk-config.json
   config-file:
     interface: content
     content: config-file
-    write: [ $SNAP_DATA/etc/deliveryoptimization-agent/sdk-config.json ]
+    write: [ $SNAP_DATA/etc/deliveryoptimization-agent/configs ]
```
Comment (shishirb-MSFT) [inline snapcraft-options/snapcraft-agent.yaml:88]: Revert the sdk-config.json change. I will do it as part of the other path fixes for DO agent.

**Review by shishirb-MSFT — CHANGES_REQUESTED** [review]

File: snapcraft-options/snapcraft-agent.yaml (hunk 6 above)
Comment (shishirb-MSFT) [reply snapcraft-options/snapcraft-agent.yaml:40]: Remove the creation of ../install/var/run/deliveryoptimization-agent

File: snapcraft-options/snapcraft-agent.yaml (hunk 7 above)
Comment (shishirb-MSFT) [reply snapcraft-options/snapcraft-agent.yaml:88]: This change has not been reverted.

File: build/build.py [hunk 8]
```diff
@@ -98,6 +98,10 @@ def __init__(self):
             '--build-for-snap', dest='build_for_snap', action='store_true',
             help='Only build components and features those required for the Ubuntu Core snap'
         )
+        self.parser.add_argument(
+            '--search-prefix', dest='search_prefix', type=str,
```
Comment (Jerry-GeGarcia) [inline build/build.py:102]: this is not related to the snap work necessarily, right?

File: snapcraft-options/snapcraft-sdk.yaml [hunk 9]
```diff
@@ -0,0 +1,76 @@
+name: deliveryoptimization-sdk
+base: core20 # the base snap is the execution environment for this snap
+version: '0.1'
+summary: Ubuntu Core 20.04 DO
+description: |
+  SDK snap to simulate and test the communication between the Delivery Optimization Client Ubuntu Core snap and the ADU Snap.
```
Comment (Jerry-GeGarcia) [inline snapcraft-options/snapcraft-sdk.yaml:6]: update to: "a consuming snap". Presumably any other snap could use the DO snap.

File: build/build.py (hunk 8 above)
Comment (shishirb-MSFT) [reply build/build.py]: Had to rebase the topic branch.

File: snapcraft-options/snapcraft-sdk.yaml (hunk 9 above)
Comment (shishirb-MSFT) [reply snapcraft-options/snapcraft-sdk.yaml]: Re-worded the comment
